src/Couchbase/Service/Couchbase.php

```diff
@@ -35,13 +35,14 @@ protected function connect()
     {
         // prepare CouchbaseCLient
         $config = $this->getConfig();
-        $client = new \Couchbase(
-            $config->get('server')->toArray(),
-            $config->get('user'),
-            $config->get('password'),
-            $config->get('bucket')
-        );
-        $this->setCouchbaseClient($client);
+        $client = new \CouchbaseCluster($config->get('server'));
+
+		$toTest = $config->get('password');
+		if(isset($toTest)) {
+			$this->setCouchbaseClient($client->openBucket($config->get('bucket'), $config->get('password')));
+		} else {
+			$this->setCouchbaseClient($client->openBucket($config->get('bucket')));
+		}
```

Comment:
This part could be shortened to return default value, if password is empty:
`$this->setCouchbaseClient($client->openBucket($config->get('bucket'), $config->get('password', '')));`

Reply:
Oh wow, completely forgot about that.
Is it still usefull ? 
If it is I'll fix that review so that it can be merged ^^